ngraph/core/include/ngraph/runtime/aligned_buffer.hpp

```diff
@@ -73,6 +75,32 @@ class NGRAPH_API ngraph::runtime::AlignedBuffer
     char* m_allocated_buffer;
     char* m_aligned_buffer;
     size_t m_byte_size;
+
+    template <typename T>
+    friend class PreallocatedBuffer;
```

Comment:
Maybe we can just move some fields to protected section instead of make a friend class.